src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -81,23 +86,26 @@
 
   public ProAI(final String name, final String type) {
     super(name, type);
+    initializeCalc();
     combatMoveAI = new ProCombatMoveAI(this);
-    nonCombatMoveAI = new ProNonCombatMoveAI();
-    purchaseAI = new ProPurchaseAI();
-    retreatAI = new ProRetreatAI();
-    scrambleAI = new ProScrambleAI();
-    politicsAI = new ProPoliticsAI();
+    nonCombatMoveAI = new ProNonCombatMoveAI(this);
+    purchaseAI = new ProPurchaseAI(this);
+    retreatAI = new ProRetreatAI(this);
+    scrambleAI = new ProScrambleAI(this);
+    politicsAI = new ProPoliticsAI(this);
     storedCombatMoveMap = null;
     storedFactoryMoveMap = null;
     storedPurchaseTerritories = null;
     storedPoliticalActions = null;
     storedStrafingTerritories = new ArrayList<Territory>();
-    initializeBattleCalculator();
   }
 
-  public void initializeBattleCalculator() {
-    final IOddsCalculator calc = new ConcurrentOddsCalculator("ProAI");
-    ProBattleUtils.setOddsCalculator(calc);
+  protected void initializeCalc() {
+    calc = new ProOddsCalculator(concurrentCalc);
+  }
+
+  public ProOddsCalculator getCalc() {
```

Comment:
So I'd prefer to pass the instance of the AI to each delegate objects and have them determine what they need from the AI. Otherwise I end up with potentially many argument constructors like I had before.